src/cable/udp/cable_udp_obj.hpp

```diff
@@ -10,23 +10,65 @@
 #include "../asiocable.hpp"
 #include "../asio_ioservice_manager.hpp"
 
+#include <cable/selector.hpp>
+#include <cable/udp/cable_udp_addr.hpp>
+
+enum class t_multisocket_kind {
+	one_rw=1, ///< one socket: for both read and write (then we do not need to e.g. do SO_REUSEPORT)
+	separate_rw, ///< two sockets: separate read, and separate write sockets, probably we need SO_REUSEPORT
+};
+
 class c_cable_udp final : public c_asiocable {
 	public:
-		c_cable_udp(shared_ptr<c_asioservice_manager> iomanager);
+		/**
+		 * create UDP cable to recv/send, using one source address (can be just port)
+		 * @param iomanager - use this manager
+		 * @param source_addr - we will send from this source address. Can be general ANY address e.g. 0.0.0.0:9042(?)
+		 * to enforce port, or a specific one to enforce entire source address.
+		*/
+		c_cable_udp(shared_ptr<c_asioservice_manager> iomanager, const c_card_selector &source_addr);
```

Comment:
:negative_squared_cross_mark: 
as above, it should be ref - thought not const-ref. made it to be ref.